service/src/main/java/uk/gov/hmcts/reform/fpl/service/PrepareDirectionsForUsersService.java

```diff
@@ -0,0 +1,152 @@
+                    }
+
+                    caseDetails.getData().put(assignee.toCustomDirectionField(), directions);
+
+                    break;
+                case OTHERS:
+                    directions.addAll(clone);
+
+                    if (eventId == COMPLY_ON_BEHALF_SDO) {
+                        filterResponsesNotCompliedOnBehalfOfByTheCourt("OTHER", directions);
+                    } else {
+                        filterResponsesNotCompliedBySolicitor(directions, OTHERS);
+                    }
+
+                    caseDetails.getData().put(assignee.toCustomDirectionField(), directions);
+
+                    break;
+                case CAFCASS:
+                    directions.addAll(clone);
+
+                    List<Element<Direction>> cafcassDirections = extractPartyResponse(COURT, directions);
+
+                    caseDetails.getData().put(assignee.toCustomDirectionField(), cafcassDirections);
+
+                    break;
+            }
+        });
+    }
+
+    private List<Element<Direction>> getClone(List<Element<Direction>> elements) {
+        return elements.stream()
+            .map(directionElement ->
+                element(directionElement.getId(), directionElement.getValue().deepCopy()))
+            .collect(toList());
+    }
+
+    /**
+     * Removes responses from a direction where they have not been complied on behalf of someone by the court.
+     *
+     * @param onBehalfOf a string matching part of the respondingOnBehalfOf variable.
+     * @param directions a list of directions.
+     */
+    public void filterResponsesNotCompliedOnBehalfOfByTheCourt(String onBehalfOf, List<Element<Direction>> directions) {
+        directions.forEach(directionElement -> directionElement.getValue().getResponses()
+            .removeIf(element -> notCompliedWithByCourt(onBehalfOf, element)));
+    }
+
+    private void filterResponsesNotCompliedBySolicitor(List<Element<Direction>> directions,
+                                                       DirectionAssignee assignee) {
+        directions.forEach(directionElement -> directionElement.getValue().getResponses()
+            .removeIf(element -> assignee != element.getValue().getAssignee()
+                || isEmpty(element.getValue().getResponder())));
+    }
+
+    private boolean notCompliedWithByCourt(String onBehalfOf, Element<DirectionResponse> element) {
+        return COURT != element.getValue().getAssignee() || isEmpty(element.getValue().getRespondingOnBehalfOf())
+            || !element.getValue().getRespondingOnBehalfOf().contains(onBehalfOf);
+    }
+
+    /**
+     * Extracts a specific response to a direction by a party.
+     *
+     * @param assignee   the role that responded to the direction.
+     * @param directions a list of directions.
+     * @return a list of directions with the correct response associated.
+     */
+    public List<Element<Direction>> extractPartyResponse(DirectionAssignee assignee,
```

Comment:
Is this supposed to be public? It's used only inside the class.

Reply:
it is, I wanted some confidence in testing as it was quite an important part of compliance to make private. Happy to lower from public